scvid/module/_probabilisticpca.py

```diff
@@ -153,16 +164,12 @@ def get_latent_representation(
         """
         Return the latent representation for each cell.
         """
-        if self.ppca_flavor == "marginalized":
-            M_kk = self.W_kg @ self.W_kg.T + self.sigma**2 * torch.eye(
-                self.k_components
-            )
-            L_gk = self.W_kg.T @ torch.linalg.inv(M_kk)
-        elif self.ppca_flavor in ("diagonal_normal", "multivariate_normal"):
-            L_gk = self.L_gk
-        else:
-            raise ValueError(
-                "ppca_flavor must be one of 'marginalized' or 'diagonal_normal' or 'multivariate_normal'"
-            )
-        z_loc_nk = (x_ng - self.mean_g) @ L_gk
+        WX_kn = self.W_kg @ (x_ng - self.mean_g).T
+        z_loc_nk = torch.linalg.solve(self.M_kk, WX_kn).T
         return z_loc_nk
+
+    @torch.inference_mode()
+    def log(self, plan: pl.LightningModule) -> None:
+        """Logging to TensorBoard by default"""
+        var_explained = torch.sum(torch.diag(self.W_kg.T @ self.W_kg) + self.sigma**2)
+        plan.log("var_explained", var_explained)
```

Comment:
This is good, but also consider calculating the total variance (from one-pass-mean-std result) as `self.total_variance` as a cached property and log the ratios (see below).  If you want to make your code agonistic to the presence of one-pass-mean-std results, this class can take a `total_variance` optional callable. If it is not None, then log the ratios as well as the unnormalized var_explained.

```
var_explained_W = torch.trace(self.W_kg.T @ self.W_kg)
plan.log("var_explained", self.sigma**2 + var_explained_W )
plan.log("var_explained_W",  var_explained_W)
plan.log("var_explained_sigma", self.sigma**2)
if self.total_variance is not None:
    plan.log("var_explained_ratio", (var_explained_W + self.sigma**2)/ self.total_variance)
    plan.log("var_explained_ratio_W",  var_explained_W / self.total_variance)
    plan.log("var_explained_ratio_sigma", self.sigma**2 / self.total_variance)
```